riff-raff/app/notification/DeployFailureNotifications.scala

```diff
@@ -50,46 +47,61 @@ class DeployFailureNotifications(config: Config,
     }
   }
 
-  def failedDeployNotification(uuid: Option[UUID], parameters: DeployParameters): Unit = {
-
-    val deriveAnghammaradTargets = for {
-      yaml <- targetResolver.fetchYaml(parameters.build)
-      deployGraph <- Resolver.resolveDeploymentGraph(yaml, deploymentTypes, magenta.input.All).toEither
-    } yield {
-      TargetResolver.extractTargets(deployGraph).toList.flatMap { target =>
-        val maybeAccountId = uuid.flatMap(uuid => getAwsAccountIdTarget(target, parameters, uuid)).toList
-        List(App(target.app), Stack(target.stack), Stage(parameters.stage.name)) ++ maybeAccountId
+  def getTargets(uuid: Option[UUID], maybeParameters: Option[DeployParameters]): List[Target] = {
+    val attemptToDeriveTargets = maybeParameters.map { parameters =>
```

Comment:
If you coerce maybeParameters into an `Either` with `toRight` then you can make this a part of the for comprehension.

This would require creating a common `Error` hierarchy so that the left hand side of the eithers match up - but this would be a good thing to do.

Reply:
I've realised that we had unnecessary optionality here, so I've managed to [simplify the function](https://github.com/guardian/riff-raff/pull/614/commits/0095af06be3764f08cb1b8ad315b52c6ef32698b) and avoid this problem entirely!